src/content/docs/infrastructure/prometheus-integrations/install-configure-prometheus-configurator/scale-prometheus-instances-instances.mdx

```diff
@@ -0,0 +1,105 @@
+
+Briefly, what the configurator does to set those relabel rules for each target is calculating a hash for the given target `__address__` and then it applies the `modulus` to the hash, being the modulus the total number of shards. Thus, it obtains into what shard the scrapped target should be included.
+
+For example, if `custom-values.yaml` includes:
+
+```yaml
+# (...)
+sharding:
+  total_shards_count: 2
+# (...)
+```
+
+And then, the release is upgraded:
+
+```shell
+helm upgrade my-prometheus-release newrelic-prometheus-configurator/newrelic-prometheus-agent -f custom-values.yaml
+```
+
+Then two prometheus servers will be executed and each target will only be scraped by only one of them.
+
+An example diagram would be the following:
+
+```mermaid
+flowchart LR
+
+    subgraph configurator[Configurator\ntotal_shards_count=2]
+    end
+
+    subgraph shards[Shards]
+        direction LR
+        desc(Each shard scrapes a subset of targets depending one the hash index\n obtained from the hashmod of the target address)
+        Shard_1(Shard_1)
+        Shard_2(Shard_2)
+    end
+
+    subgraph targets[Targets]
+        direction LR
+        Target_1
+        Target_2
+        Target_3
+    end
+
+configurator --> shards
+Shard_1 --> Target_1
+Shard_1 --> Target_3
+Shard_2 --> Target_2
+
+
+classDef plain fill:#fafafa,backgroud-color:transparent,color:#000;
+classDef text fill:#fff,stroke:transparent,color:#000,font-size:12px,font-style:italic;
+
+classDef shardy fill:#fff,stroke:transparent,color:#000,font-size:12px,font-style:italic;
+class s shardy;
+
+class desc text;
+class shards plain;
+class targets plain;
+```
+
+### Target scraper identification
+The shard identification (name of the StatefulSet Pod) is added as a `prometheus_server` label to all metrics and could be used to understand which is the Prometheus instance that is scraping that target.
+To uniquely identify a Prometheus server instance within an account a combination of `cluster_name` and `prometheus_server` labels should be used.
+
+### Self metrics
+
+Usually, prometheus server self-metrics should be gathered from all prometheus servers, so the additional rules when sharding is configured should not apply to the job gathering the prometheus self-metrics. This is possible because the _configurator_ accepts the flag `skip_sharding` in the static_target jobs. This parameter is already set up in the default self-metrics job.
+
+
+### Limitations
+
+If additional scrape jobs are included in configuration as `extra_scrape_configs`, as that field will hold the raw definition of prometheus jobs, the _configurator_ will not include the rules corresponding to sharding configuration and, as a result, the corresponding targets will be scrapped by all
+prometheus servers.
+
+Currently, auto-scaling is not supported. If increasing or decreasing the number of shards is required, an update of the cluster settings (which implies restarting the prometheus servers) is needed.
```

Comment:
```suggestion
Currently, auto-scaling is not supported. To increase or decrease the number of shards, you will need to an update of the chart settings. This will restart the prometheus pods.
```

Reply:
Thanks!